relayer/processor/path_processor.go

```diff
@@ -228,76 +210,124 @@ type pathEndPacketFlowMessages struct {
 }
 
 type pathEndConnectionHandshakeMessages struct {
+	Src                         *pathEndRuntime
+	Dst                         *pathEndRuntime
 	SrcMsgConnectionOpenInit    ConnectionMessageCache
 	DstMsgConnectionOpenTry     ConnectionMessageCache
 	SrcMsgConnectionOpenAck     ConnectionMessageCache
 	DstMsgConnectionOpenConfirm ConnectionMessageCache
 }
 
 type pathEndChannelHandshakeMessages struct {
+	Src                      *pathEndRuntime
+	Dst                      *pathEndRuntime
 	SrcMsgChannelOpenInit    ChannelMessageCache
 	DstMsgChannelOpenTry     ChannelMessageCache
 	SrcMsgChannelOpenAck     ChannelMessageCache
 	DstMsgChannelOpenConfirm ChannelMessageCache
 }
 
 type pathEndChannelCloseMessages struct {
+	Src                       *pathEndRuntime
+	Dst                       *pathEndRuntime
 	SrcMsgChannelCloseInit    ChannelMessageCache
 	DstMsgChannelCloseConfirm ChannelMessageCache
 }
 
 type pathEndPacketFlowResponse struct {
-	UnrelayedPackets          []ibcMessageWithSequence
-	UnrelayedAcknowledgements []ibcMessageWithSequence
+	SrcMessages []provider.RelayerMessage
+	DstMessages []provider.RelayerMessage
 
 	ToDeleteSrc map[string][]uint64
 	ToDeleteDst map[string][]uint64
 }
 
 type pathEndChannelHandshakeResponse struct {
-	SrcMessages []ibcMessageWithChannel
-	DstMessages []ibcMessageWithChannel
+	SrcMessages []provider.RelayerMessage
+	DstMessages []provider.RelayerMessage
 
 	ToDeleteSrc map[string][]ChannelKey
 	ToDeleteDst map[string][]ChannelKey
 }
 
 type pathEndConnectionHandshakeResponse struct {
-	SrcMessages []ibcMessageWithConnection
-	DstMessages []ibcMessageWithConnection
+	SrcMessages []provider.RelayerMessage
+	DstMessages []provider.RelayerMessage
 
 	ToDeleteSrc map[string][]ConnectionKey
 	ToDeleteDst map[string][]ConnectionKey
 }
 
-func (m *pathEndPacketFlowResponse) appendPacket(sequence uint64, msgRecvPacket provider.RelayerMessage) {
-	m.UnrelayedPackets = append(m.UnrelayedPackets, ibcMessageWithSequence{Sequence: sequence, Message: msgRecvPacket})
+// assembleIBCMessage constructs the applicable IBC message using the requested function.
+// These functions may do things like make queries in order to assemble a complete IBC message.
+func (pp *PathProcessor) assembleIBCMessage(
+	ctx context.Context,
+	src, dst *pathEndRuntime,
+	message string,
+	partialMessage provider.RelayerMessage,
+	assembleMessage func(ctx context.Context, msgRecvPacket provider.RelayerMessage, signer string, latest provider.LatestBlock) (provider.RelayerMessage, error),
+	messages *[]provider.RelayerMessage,
+) error {
+	signer, err := dst.chainProvider.Address()
+	if err != nil {
+		return fmt.Errorf("error getting signer address for {%s}: %w\n", dst.info.ChainID, err)
```

Comment:
You could omit the newlines at the end of these errors.

Reply:
good catch